server/src/main/java/org/elasticsearch/common/geo/GeometryTreeWriter.java

```diff
@@ -110,9 +139,7 @@ private void addWriter(ShapeTreeWriter writer) {
 
         @Override
         public Void visit(GeometryCollection<?> collection) {
-            for (Geometry geometry : collection) {
-                geometry.visit(this);
-            }
+            addWriter(new GeometryTreeWriter(collection, coordinateEncoder));
```

Comment:
I looked at the GeoJSON spec, and it does not look like a geometrycollection can have a nested geometrycollection within it. I am not sure about WKT. do you know?

Reply:
I think it is not as strong. According to RFC 7946

> To maximize interoperability, implementations SHOULD avoid nested GeometryCollections.

We support it elsewhere, so if we want to stop supporting it we should depreciate it first in other places.

Reply:
oh OK. good to know!